src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -353,10 +356,12 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
                     // Don't run set for a parameter passed to the constructor
                     $params[] = $parameterData;
                     unset($data[$key]);
+                } elseif ($allowed && !$ignored && isset($context[static::EMPTY_DATA][$class][$key])) {
```

Comment:
I would not check for `$allowed && !$ignored`, because this data is provided by the developer, not by the end user.

Reply:
It can lead to weird behaviors but I'm in favor of given more power to the developer too. 👍